### PR TITLE
Support 0d/1d Mat in FileStorage

### PR DESCRIPTION
**Pull request by dkurt**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by vpisarev** [comment]
@dkurt, this is definitely useful!

I wonder, if you write to FileStorage an empty mat:

```
fs << "empty-mat" << Mat();
```

whether it will be done correctly? I mean, 0-dimensional matrix can mean two different things in OpenCV, a scalar or an empty matrix, depending on whether `data == nullptr` or not (can use `Mat::empty()` method as well).

**Comment by dkurt** [comment]
@vpisarev, added a test for empty Mat and required changes to handle it properly.

**Review by asmorkalov — APPROVED** [review]

File: modules/core/test/test_io.cpp [hunk 1]
```diff
@@ -2025,4 +2025,49 @@ TEST(Core_InputOutput, FileStorage_invalid_attribute_value_regression_25946)
...
+    bool srcIsEmpty = src.empty();
+    Mat dst = fsWriteRead(src, ext);
+    EXPECT_EQ(dst.empty(), srcIsEmpty);
+    EXPECT_EQ(src.dims, dst.dims);
+    EXPECT_EQ(src.size, dst.size);
+    EXPECT_EQ(cv::norm(src, dst, NORM_INF), 0.0);
+}
+
+typedef testing::TestWithParam<const char*> FileStorage_exact_type;
+TEST_P(FileStorage_exact_type, empty_mat)
```
Comment (mshabunin) [inline modules/core/test/test_io.cpp:2054]: This test raises debug assertion
```
[ RUN      ] Core_InputOutput/FileStorage_exact_type.empty_mat/0, where GetParam() = ".yml"
unknown file: Failure
C++ exception with description "OpenCV(5.0.0-pre) /work/opencv/modules/core/include/opencv2/core/mat.inl.hpp:1349: error: (-215:Assertion failed) i < dims() in function 'operator[]'
" thrown in the test body.
[  FAILED  ] Core_InputOutput/FileStorage_exact_type.empty_mat/0, where GetParam() = ".yml" (1 ms)
```
Here:
https://github.com/opencv/opencv/blob/979428d5908b5396237b18cbd50d0b18265e2463/modules/core/include/opencv2/core/mat.inl.hpp#L1346-L1354

Comment (dkurt) [reply modules/core/test/test_io.cpp:2054]: Same on 4.x branch, thanks. I will check

**Pull request merged**